esphome/components/adc/adc_sensor_esp32.cpp

```diff
@@ -78,12 +78,14 @@ void ADCSensor::dump_config() {
     }
   }
   ESP_LOGCONFIG(TAG, "  Samples: %i", this->sample_count_);
+  ESP_LOGCONFIG(TAG, "  Sampling mode: %s", sampling_mode_str(this->sampling_mode_));
```

Comment:
```suggestion
  ESP_LOGCONFIG(TAG, "  Sampling mode: %s", LOG_STR_ARG(sampling_mode_to_str(this->sampling_mode_)));
```
https://github.com/esphome/esphome/blob/fc847c1de8927871e783db1a35e967efabda9f04/esphome/core/log.h#L168-L194